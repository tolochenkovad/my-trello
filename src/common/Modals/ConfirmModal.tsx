import React, { FC } from 'react';
import { Modal } from 'react-bootstrap';
import Buttons from './Buttons';

type Props = {
  show: boolean;
  onHide: () => void;
  onConfirm: () => void;
};

const ConfirmModal: FC<Props> = ({ show, onHide, onConfirm }) => (
  <Modal show={show} onHide={onHide} backdrop="static">
    <Modal.Body>
      <div className="text-center font-size-20">Are you sure?</div>
    </Modal.Body>
    <Modal.Footer className="no-border">
      <Buttons onHide={onHide} onConfirm={onConfirm} />
    </Modal.Footer>
  </Modal>
);

export default ConfirmModal;
