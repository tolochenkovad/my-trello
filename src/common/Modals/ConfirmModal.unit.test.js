import React from 'react';
import { shallow } from 'enzyme';
import ConfirmModal from './ConfirmModal';

const props = {
  show: false,
  onHide: jest.fn(),
};

let wrapper;

beforeEach(() => {
  wrapper = shallow(<ConfirmModal {...props} />);
});

describe('ConfirmModal', () => {
  it('should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('children is exist', () => {
    it('Modal', () => {
      expect(wrapper.find('Modal')).toHaveLength(1);
    });

    it('body', () => {
      expect(wrapper.find('ModalBody')).toHaveLength(1);
    });

    it('footer', () => {
      expect(wrapper.find('ModalFooter')).toHaveLength(1);
    });
  });

  it('should call onHide when click on close button', () => {
    expect(wrapper.props().onHide).not.toHaveBeenCalled();
    wrapper.find('Button').simulate('click');
    expect(wrapper.props().onHide).toHaveBeenCalled();
  });
});
