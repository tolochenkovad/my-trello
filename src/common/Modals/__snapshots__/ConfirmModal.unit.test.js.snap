// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmModal should render correctly 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop="static"
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "displayName": "ModalDialog",
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={false}
  onHide={[MockFunction]}
  restoreFocus={true}
  show={false}
>
  <ModalBody>
    <div
      className="text-center"
    >
      Are you sure?
    </div>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[MockFunction]}
      variant="primary"
    >
      Confirm
    </Button>
  </ModalFooter>
</Modal>
`;
